File: Modul-06/SOAL 5/PRAK605-2310817210009-MuhammadIbnuSina.py
def inputMatriks(ordo):
    matriks = []
    for i in range(ordo):
        baris = list(map(int, input().split()))
        matriks.append(baris)
    return matriks

def multiplyMatrices(ordo, matriksA, matriksB):
    result = [[0] * ordo for _ in range(ordo)]
    for i in range(ordo):
        for j in range(ordo):
            result[i][j] = 0
            for x in range(ordo):
                result[i][j] += matriksA[i][x] * matriksB[x][j]
    return result

def printMatriks(ordo, matriks):
    print("Matriks AXB:")
    for i in range(ordo):
        for j in range(ordo):
            print(matriks[i][j], end=" ")
        print()

def main():
    ordo = int(input())

    print("Matriks A:")
    matriksA = inputMatriks(ordo)

    print("Matriks B:")
    matriksB = inputMatriks(ordo)

    matriksAXB = multiplyMatrices(ordo, matriksA, matriksB)

    print()
    printMatriks(ordo, matriksAXB)

if __name__ == "__main__":
    main()

File: Modul-06/SOAL 5/PRAK605-2310817210009-MuhammadIbnuSina.c
#include <stdio.h>

void inputMatriks(int ordo, int matriks[ordo][ordo]) {
    for (int i = 0; i < ordo; i++) {
        for (int j = 0; j < ordo; j++) {
            scanf("%d", &matriks[i][j]);
        }
    }
}

void multiplyMatrices(int ordo, int matriksA[ordo][ordo], int matriksB[ordo][ordo], int result[ordo][ordo]) {
    for (int i = 0; i < ordo; i++) {
        for (int j = 0; j < ordo; j++) {
            result[i][j] = 0;
            for (int x = 0; x < ordo; x++) {
                result[i][j] += matriksA[i][x] * matriksB[x][j];
            }
        }
    }
}

void printMatriks(int ordo, int matriks[ordo][ordo]) {
    printf("Matriks AXB:\n");
    for (int i = 0; i < ordo; i++) {
        for (int j = 0; j < ordo; j++) {
            printf("%d ", matriks[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int ordo;
    scanf("%d", &ordo);

    int matriksA[ordo][ordo];
    int matriksB[ordo][ordo];
    int matriksAXB[ordo][ordo];

    printf("Matriks A\n");
    inputMatriks(ordo, matriksA);

    printf("Matriks B\n");
    inputMatriks(ordo, matriksB);

    multiplyMatrices(ordo, matriksA, matriksB, matriksAXB);

        printf("\n");
    printMatriks(ordo, matriksAXB);

    return 0;
}